File: src/pi.php
<?php
function calculatePi($precision)
{
  $s = 0.0;
  $k = 1.0;
  $max = pow(10, $precision);
  for ($i = 0; $i < $max; $i++) {
    if ($i % 2 == 0)
      $s += 4 / $k;
    else
      $s -= 4 / $k;
    $k += 2;
  }
  return $s;
}

$pi = calculatePi(9);
printf("Pi = %.8f\n", $pi);

File: src/pi.rs
fn calculate_pi(precision: i32) -> f64 {
  let (mut pi, mut k): (f64, f64) = (0.0, 1.0);
  let max: i32 = i32::pow(10, precision as u32);
  for i in 0..max {
    if i % 2 == 0 {
      pi = pi + 4.0 / k;
    } else {
      pi = pi - 4.0 / k;
    }
    k = k + 2.0;
  }
  return pi;
}

fn main() {
  let pi: f64 = calculate_pi(9);
  println!("Pi = {:.8}", pi)
}
